src/sensai/vector_model.py

```diff
@@ -334,6 +247,23 @@ def predict(self, x: pd.DataFrame) -> pd.DataFrame:
     def _predict(self, x: pd.DataFrame) -> pd.DataFrame:
         pass
 
+    def _underlyingModelRequiresFitting(self) -> bool:
+        """
+        Designed to be overridden for rule-based models.
+
+        :return: True iff the underlying model requires fitting
```

Comment:
though I like mathematical notation, here "if" with one f will be enough

Reply:
Not really. If one "f" is removed, then the specification is incomplete. For it to be complete, one would need to add ", False otherwise".
Why do you want me to change this?